File: 2018_Graduation_project#convert_to_java/TestApp/app/src/main/java/com/android/test/DatabaseHelper.java
package com.android.test;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.database.sqlite.SQLiteQueryBuilder;

import java.util.ArrayList;

public final class DatabaseHelper extends SQLiteOpenHelper {
    private Context context = this.getContext();
    private SQLiteDatabase sqlObj = this.getWritableDatabase();
    private static final String DBName = "MenuDB";
    private static final int DBversion = 2;
    private static final String TableName = "MenuTable";
    private static final String PrimaryKey = "RowId";
    private static final String FirstColumn = "TabName";
    private static final String SecondColumn = "MenuName";
    private static final String ThirdColumn = "Price";
    private static final String FourthColumn = "Image";

    public DatabaseHelper(Context context) {
        super(context, DBName, (SQLiteDatabase.CursorFactory)null, DBversion);
        this.context = context;
        SQLiteDatabase sqlObj= this.getWritableDatabase();
        this.sqlObj = sqlObj;
    }

    public final Context getContext() { return this.context; }

    public void onCreate(SQLiteDatabase DB) {
        String CREATE_TABLE_SQL = "CREATE TABLE IF NOT EXISTS " + TableName +
                " " + "(" + PrimaryKey + " INTEGER PRIMARY KEY," + FirstColumn + " TEXT," +
                SecondColumn + " TEXT," + ThirdColumn + " INTEGER," + FourthColumn + " BLOB);";
        DB.execSQL(CREATE_TABLE_SQL);
    }

    public void onUpgrade(SQLiteDatabase DB, int oldVersion, int newVersion) {
        DB.execSQL("Drop table IF EXISTS " + TableName);
        this.onCreate(DB);
    }

    public final String InsertMenu(ContentValues values) {
        String Msg = "error";
        long ID = sqlObj.insert(TableName, (String)null, values);
        if (ID > 0L) {
            Msg = "ok";
        }
        return Msg;
    }

    public final String RemoveMenu(int RowId) {
        String[] selectionArs = new String[]{String.valueOf(RowId)};
        int i = sqlObj.delete(TableName, "RowId=?", selectionArs);
        return i > 0 ? "ok" : "error";
    }

    public final int RemoveAll() {
        return sqlObj.delete(TableName, (String)null, (String[])null);
    }

    public final String UpdateMenu(ContentValues values, int RowId) {
        String[] selectionArgs = new String[]{String.valueOf(RowId)};
        int i = sqlObj.update(TableName, values, "RowId=?", selectionArgs);
        return i > 0 ? "ok" : "error";
    }

    public final int RemoveTab(String TabName) {
        String[] selectionArgs = new String[]{TabName};
        return sqlObj.delete(TableName, "TabName=?", selectionArgs);
    }

    public final ArrayList FetchTabName() {
        ArrayList arraylist = new ArrayList();
        SQLiteQueryBuilder sqb = new SQLiteQueryBuilder();
        sqb.setTables(TableName);
        sqb.setDistinct(true);
        String[] cols = new String[]{"TabName"};
        Cursor cur = sqb.query(this.sqlObj, cols, (String)null, (String[])null, (String)null, (String)null, "TabName");
        if (cur.moveToFirst()) {
            do {
                String TabName = cur.getString(cur.getColumnIndex("TabName"));
                arraylist.add(TabName);
            } while(cur.moveToNext());
        }

        return arraylist;
    }

    public final ArrayList FetchMenu(String TabName) {
        ArrayList arraylist = new ArrayList();
        SQLiteQueryBuilder sqb = new SQLiteQueryBuilder();
        sqb.setTables(TableName);
        String[] cols = new String[]{"RowId", "MenuName", "Price", "Image"};
        String[] SelArg = new String[]{TabName};
        Cursor cur = sqb.query(this.sqlObj, cols, "TabName like ?", SelArg, (String)null, (String)null, "TabName");
        if (cur.moveToFirst()) {
            do {
                int RowId = cur.getInt(cur.getColumnIndex("RowId"));
                String MenuName = cur.getString(cur.getColumnIndex("MenuName"));
                int Price = cur.getInt(cur.getColumnIndex("Price"));
                byte[] Image = cur.getBlob(cur.getColumnIndex("Image"));
                MenuData temp = new MenuData(TabName, MenuName, Price);
                if (Image != null) {
                    temp.setImage(Image);
                }

                temp.setRowId(RowId);
                arraylist.add(temp);
            } while(cur.moveToNext());
        }

        return arraylist;
    }

    public final String FindName(int rowid){
        String returnname = new String();
        SQLiteQueryBuilder sqb = new SQLiteQueryBuilder();
        sqb.setTables(TableName);
        String[] cols = new String[]{"RowId", "MenuName", "Price", "Image"};
        String[] SelArg = new String[]{Integer.toString(rowid)};
        Cursor cur = sqb.query(this.sqlObj, cols, "RowId like ?", SelArg, (String)null, (String)null, null);
        if (cur.moveToFirst()) {
            do {
                returnname = cur.getString(cur.getColumnIndex("MenuName"));
            } while(cur.moveToNext());
        }
        return returnname;
    }


}

File: 20180609_android_kotlin/app/src/main/java/com/example/master/a20180609_android_kotlin/DatabaseHelper.kt
package com.example.master.a20180609_android_kotlin

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import android.database.sqlite.SQLiteQueryBuilder
import android.media.Image

class DatabaseHelper (context: Context) : SQLiteOpenHelper(context, DBName, null, DBversion) {

    companion object {
        val DBName = "MenuDB"
        val DBversion = 2

        val TableName = "MenuTable"
        val PrimaryKey = "RowId"

        val FirstColumn = "TabName"
        val SecondColumn = "MenuName"
        val ThirdColumn = "Price"
        val FourthColumn = "Image"
    }

    var context: Context = context
    var sqlObj: SQLiteDatabase = this.writableDatabase

    override fun onCreate(DB: SQLiteDatabase?) {
        var CREATE_TABLE_SQL : String = "CREATE TABLE IF NOT EXISTS " + TableName + " " +
                "(" + PrimaryKey + " INTEGER PRIMARY KEY," + FirstColumn + " TEXT," + SecondColumn + " TEXT," + ThirdColumn + " INTEGER," + FourthColumn + " BLOB);"
        DB!!.execSQL(CREATE_TABLE_SQL)
    }

    override fun onUpgrade(DB: SQLiteDatabase?, oldVersion : Int, newVersion: Int) {
        DB!!.execSQL("Drop table IF EXISTS "+ TableName)
        onCreate(DB)
    }

    fun InsertMenu(values: ContentValues): String{
        var Msg: String = "error"
        val ID = sqlObj!!.insert(TableName,null,values)

        if(ID > 0)
            Msg = "ok"
        return Msg
    }

    fun RemoveMenu(RowId : Int): String {

        var selectionArs = arrayOf(RowId.toString())

        val i = sqlObj!!.delete(TableName, "RowId=?", selectionArs)
        if (i > 0) {
            return "ok";
        } else {
            return "error";
        }
    }

    fun RemoveAll() : Int {
        return sqlObj!!.delete(TableName,null,null)
    }

    fun UpdateMenu(values: ContentValues, RowId: Int): String {

        var selectionArs = arrayOf(RowId.toString())

        val i = sqlObj!!.update(TableName, values, "RowId=?", selectionArs)
        if (i > 0) {
            return "ok";
        } else {
            return "error";
        }
    }

    fun FetchTabName(): ArrayList<String>{
        var arraylist = ArrayList<String>()

        val sqb = SQLiteQueryBuilder()
        sqb.tables = TableName
        sqb.setDistinct(true)
        val cols = arrayOf("TabName")

        val cur = sqb.query(sqlObj, cols, null, null, null, null, "TabName")

        if (cur.moveToFirst()) {
            do {
                val TabName = cur.getString(cur.getColumnIndex("TabName"))
                arraylist.add(TabName)
            } while (cur.moveToNext())
        }

        return arraylist
    }

    fun FetchMenu(TabName : String): ArrayList<MenuData> {

        var arraylist = ArrayList<MenuData>()

        val sqb = SQLiteQueryBuilder()
        sqb.tables = TableName
        val cols = arrayOf("RowId", "TabName", "MenuName", "Price", "Image")
        val SelArg = arrayOf(TabName)

        val cur = sqb.query(sqlObj, cols, "TabName like ?", SelArg, null, null, "TabName")

        if (cur.moveToFirst()) {
            do {
                val RowId = cur.getInt(cur.getColumnIndex("RowId"))
                val TabName = cur.getString(cur.getColumnIndex("TabName"))
                val MenuName = cur.getString(cur.getColumnIndex("MenuName"))
                val Price = cur.getInt(cur.getColumnIndex("Price"))
                val Image = cur.getBlob(cur.getColumnIndex("Image"))

                val temp:MenuData = MenuData(TabName,MenuName,Price)
                if(Image != null) {
                    temp.Image = Image
                }
                temp.RowId = RowId
                arraylist.add(temp)

            } while (cur.moveToNext())
        }
        return arraylist
    }


}